compat/memset_s.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_MEMSET_S_H
+#define SQUID_COMPAT_MEMSET_S_H
+
+#if !HAVE_MEMSET_S
+
+typedef int errno_t;
+typedef size_t rsize_t;
+
+/**
+ * to zero a buffer in a more secure manner meant for a handful of purposes.
+ * e.g. for password matters.
+ * The compiler can optimize away a memset call to gain performance here
+ * making sure it does not occur.
+ *
+ * memset_s is designed for such cases otherwise putting the
+ * address in a volatile pointer avoid gcc's likes doing optimizations.
```

Comment:
This is an implementation-specific detail. Please move it to the code implementing it (not the whole function definition where it is currently duplicated, but the actual volatile trick).